### PR TITLE
Fixes the dreaded cloning bug

### PR DESCRIPTION
**Pull request by Time-Green**
This took me way to long.
Fixes a bug that would not clone gibbed people, and in other edge cases.
Closes: https://github.com/yogstation13/yogstation/issues/2197
Closes: https://github.com/yogstation13/yogstation/issues/2045

:cl:
fix: Fixes not getting placed in your new cloned body in certain events
/:cl:

### CODE REVIEW

**Comment by Time-Green** [comment]
There appears to be a problem with it forgetting antag status...

**Comment by Cruix** [comment]
See if the mind gets deleted when the mob does. If it does not, you can keep a reference to it in the cloner to keep it from being garbage collected.

**Comment by Time-Green** [comment]
The problem was it actually deleting the mind with it, so the reference was useless since it made a new one. The antag stuff and everything goes away with it. I am currently trying to prevent it from actually deleting the mind the way it does

**Comment by Time-Green** [comment]
Alternatively...
We could make a copy of the mind datum and keep that stored. And edit some things to make it work right.
This has the fun side effect of getting completely reverted to the point you were scanned, so if you got antag after getting scanned, it would be gone. Or the other way around.

But there is the side effect of people abusing it to use one-time powers like destroy engines infinitely

**Comment by Time-Green** [comment]
Or now I think of it, just use the old mind and put the player inside that, fixing all problems..

**Comment by matskuman5** [comment]
ah, beautiful work time, i get so many bwoinks about bugged cloning

**Comment by Time-Green** [comment]
I'll fix it tomorrow. I already have it, but it hasn't been tested

**Comment by ShadowDeath6** [comment]
I think destroy engines works on a global var so other shadowlings can't use it

**Comment by Time-Green** [comment]
Nah, every single one can us em
Anyway, I think I fixed it without doing the old mind thing

File: code/game/machinery/cloning.dm [hunk 1]
```diff
@@ -342,9 +342,10 @@
 		//Don't dump us into clone if we somehow got a body during cloning finished
 		if(clonemind.current && clonemind.current.stat == DEAD)
 			clonemind.transfer_to(occupant)
-		occupant.grab_ghost()
-		occupant << "<span class='notice'><b>The world is suddenly bright \
-			and sudden and loud!</b><br>\
+		else
+			occupant.ckey = clonemind.key
+		occupant << "<span class='notice'><b>The world is sudden, bright \
```
Comment (ShadowDeath6) [inline code/game/machinery/cloning.dm]: please make this less terrible, anything else that isn't this sentence will do

Comment (Time-Green) [reply code/game/machinery/cloning.dm]: I already changed it, but I need to test it

**Comment by Time-Green** [comment]
Done. It works exactly as intended now. It clones the way it's supposed to and also keeps the mind

**Comment by ShadowDeath6** [comment]
Can you fix / remove the sudden and loud message you get on cloning it's so awful

"You feel dull pain as your newly formed body is dropped out of [pod], exposing it to the outside elements." Or anything that doesn't involve the word sudden

**Comment by ShadowDeath6** [comment]
thanks

**Comment by Cruix** [comment]
Did you fix the mind datum being deleted when the body is destroyed?

**Comment by Time-Green** [comment]
No, it instead recovers the one from the initial scan

**Pull request merged**